aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -0,0 +1,614 @@
+        "ASPNETCORE_ENVIRONMENT": "Development",
+        "ASPNETCORE_URLS": "http://localhost:5000/;http://localhost:5001/",
+        "ASPNETCORE_MANAGEMENTPORT": "5001"
+      },
+      "applicationUrl": "http://localhost:5000/"
+    }
+  }
+}
+```
+
+Register health check services with `AddHealthChecks` in `Startup.ConfigureServices`. The call to `UseHealthChecks` specifies the management port (*ManagementPortStartup.cs*):
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/ManagementPortStartup.cs?name=snippet1&highlight=12,18)]
+
+> [!NOTE]
+> You can avoid creating the *launchSettings.json* file in the sample app by setting the URLs and management port explicitly in code. In *Program.cs* where the `WebHostBuilder` is created, add a call to `UseUrls` and provide the app's normal response endpoint and the management port endpoint. In *ManagementPortStartup.cs* where `UseHealthChecks` is called, specify the management port explicitly.
+>
+> *Program.cs*:
+>
+> ```csharp
+> return new WebHostBuilder()
+>     .UseConfiguration(config)
+>     .UseUrls("http://localhost:5000/;http://localhost:5001/")
+>     .ConfigureLogging(builder =>
+>     {
+>         builder.SetMinimumLevel(LogLevel.Trace);
+>         builder.AddConfiguration(config);
+>         builder.AddConsole();
+>     })
+>     .UseKestrel()
+>     .UseStartup(startupType)
+>     .Build();
+> ```
+>
+> *ManagementPortStartup.cs*:
+>
+> ```csharp
+> app.UseHealthChecks("/health", port: 5001);
+> ```
+
+To run the management port configuration scenario using the sample app, execute the following command from the project's folder in a command shell:
+
+```console
+dotnet run --scenario port
+```
+
+## Distribute a health check library
+
+To distribute a health check as a library:
+
+1. Write a health check that implements the `IHealthCheck` interface as a standalone class. The class can rely on [dependency injection (DI)](xref:fundamentals/dependency-injection), type activation, and [named options](xref:fundamentals/configuration/options) to access configuration data.
+
+   ```csharp
+   using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+   public class ExampleHealthCheck : IHealthCheck
+   {
+       private readonly string _data1;
+       private readonly int? _data2;
+
+       public SqlServerHealthCheck(string data1, int? data2)
+       {
+           _data1 = data1 ?? throw new ArgumentNullException(nameof(data1));
+           _data2 = data2 ?? throw new ArgumentNullException(nameof(data2));
+       }
+       public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+       {
+           try
+           {
+               // Health check logic
+               //
+               // data1 and data2 are used in the method to
+               // run the probe's health check logic.
+
+               return HealthCheckResult.Healthy();
+           }
+           catch (Exception ex)
```

Comment:
We wouldn't recommend that a health check handle all exceptions by default. We'll do this for you.

If you do expect the code you're calling inside the check to throw exceptions, then handle it and return `context.Registration.FailureStatus`.

Either way, you should indictate a failure by returning `context.Registration.FailureStatus` from a library. This allows users of your library to configure the severity of the failure.